src/libs/ReportUtils.ts

```diff
@@ -1688,102 +1478,87 @@ function getModifiedExpenseMessage(reportAction) {
     // to match how we handle the modified expense action in oldDot
-    if (_.has(transactionChanges, 'amount') || _.has(transactionChanges, 'currency')) {
+    if (Object.prototype.hasOwnProperty.call(transactionChanges, 'amount') || Object.prototype.hasOwnProperty.call(transactionChanges, 'currency')) {
         originalMessage.oldAmount = TransactionUtils.getAmount(oldTransaction, isFromExpenseReport);
-        originalMessage.amount = lodashGet(transactionChanges, 'amount', originalMessage.oldAmount);
+        originalMessage.amount = transactionChanges?.amount.originalMessage.oldAmount;
         originalMessage.oldCurrency = TransactionUtils.getCurrency(oldTransaction);
-        originalMessage.currency = lodashGet(transactionChanges, 'currency', originalMessage.oldCurrency);
+        originalMessage.currency = transactionChanges?.currency.originalMessage.oldCurrency;
     }
 
-    if (_.has(transactionChanges, 'category')) {
+    if (Object.prototype.hasOwnProperty.call(transactionChanges, 'category')) {
         originalMessage.oldCategory = TransactionUtils.getCategory(oldTransaction);
-        originalMessage.category = transactionChanges.category;
+        originalMessage.category = transactionChanges?.category;
     }
 
-    if (_.has(transactionChanges, 'tag')) {
+    if (Object.prototype.hasOwnProperty.call(transactionChanges, 'tag')) {
         originalMessage.oldTag = TransactionUtils.getTag(oldTransaction);
-        originalMessage.tag = transactionChanges.tag;
+        originalMessage.tag = transactionChanges?.tag;
     }
 
-    if (_.has(transactionChanges, 'billable')) {
+    if (Object.prototype.hasOwnProperty.call(transactionChanges, 'billable')) {
         const oldBillable = TransactionUtils.getBillable(oldTransaction);
         originalMessage.oldBillable = oldBillable ? Localize.translateLocal('common.billable').toLowerCase() : Localize.translateLocal('common.nonBillable').toLowerCase();
-        originalMessage.billable = transactionChanges.billable ? Localize.translateLocal('common.billable').toLowerCase() : Localize.translateLocal('common.nonBillable').toLowerCase();
+        originalMessage.billable = transactionChanges?.billable ? Localize.translateLocal('common.billable').toLowerCase() : Localize.translateLocal('common.nonBillable').toLowerCase();
     }
 
     return originalMessage;
 }
 
 /**
  * Returns the parentReport if the given report is a thread.
- *
- * @param {Object} report
- * @returns {Object}
  */
-function getParentReport(report) {
-    if (!report || !report.parentReportID) {
+function getParentReport(report: OnyxEntry<Report>): OnyxEntry<Report> | undefined | Record<string, never> {
+    if (!report?.parentReportID) {
         return {};
     }
-    return lodashGet(allReports, `${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID}`, {});
+    return allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID}`];
```

Comment:
```suggestion
    return allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${report.parentReportID}`] ?? {};
```
This way you can also remove undefined from return type